Source/ProgressionSystemRuntime/Private/Widgets/PSOverlayWidget.cpp

```diff
@@ -120,27 +111,41 @@ void UPSOverlayWidget::TickPlayFadeOverlayAnimation()
 // When a character has been changed current active progression row also changes
 void UPSOverlayWidget::OnCurrentRowDataChanged_Implementation(FPlayerTag PlayerTag)
 {
-	DisplayLevelUIOverlay();
-}
-
-void UPSOverlayWidget::SetOverlayItemsVisibility(ESlateVisibility VisibilitySlate)
-{
-	// Level is unlocked hide the blocking overlay
-	SetVisibility(VisibilitySlate);
+	const bool bIsNewCharacter = PreviousPlayerTagInternal != PlayerTag;
+	DisplayLevelUIOverlay(bIsNewCharacter);
+	PreviousPlayerTagInternal = PlayerTag;
 }
 
 // Show or hide the LevelUIOverlay depends on the level lock state for current level
 // by default overlay is always displayed 
-void UPSOverlayWidget::DisplayLevelUIOverlay()
+void UPSOverlayWidget::DisplayLevelUIOverlay(bool bIsnewCharacter)
```

Comment:
`bIsnewCharacter` please make it in camel case with capital `New`, but not `new`

Reply:
done https://github.com/moinm3uw/ProgressionSystem/pull/63/commits/7df055e5a971cf2605e24a95d37e033e96aa7126